proxy/src/cancellation.rs

```diff
@@ -367,8 +366,7 @@ impl CancellationHandler {
             return Err(CancelError::InternalError);
         };
 
-        tx.send_timeout(op, REDIS_SEND_TIMEOUT)
-            .await
+        tx.try_send(op)
```

Comment:
This change seems to be the much bigger one. Now we drop `op`s immediately when the buffer is full instead of waiting for a bit. Make the PR about this.